### PR TITLE
Add and configure scheduled transitions module.

### PR DESCRIPTION
**Pull request by weitzman**
**Description:**
- See Jira ticket

**Jira:** (Skip unless you are MA staff)
DP-19428

**To Test:**
- [ ] See Jira ticket for change listing

**Follow-up:**
- Once this is merged, start work on a new ticket to disable scheduler modules. I would have done that here but it would make code review cumbersome (too much scope in one PR).

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: docroot/modules/custom/mass_caching/tests/src/ExistingSite/AutomatedPurgingTest.php [hunk 1]
```diff
@@ -65,10 +65,12 @@ public function testAliasCreationResultsInPurge() {
         'alias' => '/foo-foo',
       ],
     ]);
-    $this->assertCount(1, $this->getInvalidations('url', 'http://mass.local/foo-foo'));
+    $url = $node->toUrl('canonical', ['absolute' => TRUE])->toString();
+    $this->assertCount(1, $this->getInvalidations('url', $url));
```
Comment (weitzman) [inline docroot/modules/custom/mass_caching/tests/src/ExistingSite/AutomatedPurgingTest.php:70]: These changes help tests pass when using an alternate domain (i.e. on native dev env).

File: docroot/modules/custom/mayflower/mayflower.info.yml [hunk 2]
```diff
@@ -3,3 +3,5 @@ type: module
 description: Glue mayflower Pattern labs templates to Drupal entities.
 core: 8.x
 package: Custom
+dependencies:
+  - components
```
Comment (weitzman) [inline docroot/modules/custom/mayflower/mayflower.info.yml:8]: A tiny unrelated improvement

**Comment by arthurbaghdas** [comment]
Composer install fails with an error "Cannot apply patch Fixes field label when a new entity is created inline (h  
  ttps://www.drupal.org/project/inline_entity_form/issues/2842744) (https://w  
  ww.drupal.org/files/issues/inline_entity_form-no_label_required_field_with_  
  no_entries-2842744.patch)! "
  
  There are conflicts in the composer.lock file
  
  @weitzman

**Comment by weitzman** [comment]
Composer issues are resolved.

File: docroot/modules/custom/mass_scheduled_transitions/mass_scheduled_transitions.module [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+  /** @var \Drupal\scheduled_transitions\Entity\ScheduledTransition $transition */
+  $transition = ScheduledTransition::create([]);
+  $datetime = (new DrupalDateTime("now + $duration"))->getPhpDateTime();
+  $transition->setTransitionDate($datetime)
+    ->setAuthor(Drupal::currentUser())
+    ->setState(Workflow::load($workflow), MassModeration::UNPUBLISHED)
+    ->setEntity($node)
+    // 0 means 'latest'.
+    ->setEntityRevisionId(0)
+    ->setOptions([MASS_SCHEDULED_TRANSITIONS_OPTIONS])
```
Comment (larowlan) [inline docroot/modules/custom/mass_scheduled_transitions/mass_scheduled_transitions.module:89]: is the nested array correct here, the const is an array too?

Comment (larowlan) [reply docroot/modules/custom/mass_scheduled_transitions/mass_scheduled_transitions.module:89]: i guess that's a peculiarity of map fields

**Pull request merged**